Neos.Flow/Classes/Validation/Validator/GenericObjectValidator.php

```diff
@@ -43,6 +48,15 @@ public function setValidatedInstancesContainer(\SplObjectStorage $validatedInsta
         $this->validatedInstancesContainer = $validatedInstancesContainer;
     }
 
+    /**
+     * @param object $object The object instance to register a validation callback for
+     * @param \Closure $callback The callback to invoke once this object has been validated
+     */
+    public static function onValidated(object $object, \Closure $callback)
```

Comment:
Static method callback registration. It's a special case, so signal slot didn't seem like the right approach. We'd need to register a single slot every time and invoke it for every object validation, not only `iff` a callback for the specific object instance is registered.

So why on the `GenericObjectValidator`? Because it's the root validator for every object type, so it will be invoked for every object that gets validated (see https://github.com/neos/flow-development-collection/blob/5.3/Neos.Flow/Classes/Validation/ValidatorResolver.php#L291-L296). The `AggregateBoundaryValidator` extends this and will just early out for unloaded proxies then delegate to this.